src/core/tests/type_prop/if.cpp

```diff
@@ -340,6 +340,36 @@ TEST(type_prop, if_scalar_and_1d_static_union) {
     EXPECT_EQ(sh, out_shape);
 }
 
+TEST(type_prop, if_output_one_element) {
+    // That which we iterate over
+    auto X = make_shared<ov::op::v0::Parameter>(element::f32, Shape{});
+    auto Y = make_shared<ov::op::v0::Parameter>(element::f32, Shape{1});
```

Comment:
With current changes, the test will pass for Y Parameter with dynamic shape `PartialShape{Dimension(0, 1)}`, so the inferred output shape from the `If` operation will be static `PartialShape{1}`.